3.9-minimal/README.md

```diff
@@ -1,117 +1,61 @@
-Building a simple [python-sample-app](https://github.com/sclorg/django-ex.git) application
-in Openshift can be achieved with the following step:
-
-    ```
-    oc new-app python:3.9~https://github.com/sclorg/django-ex.git
-    ```
-
-**Accessing the application:**
-```
-$ oc get pods
-$ oc exec <pod> -- curl 127.0.0.1:8080
-```
-
-Source-to-Image framework and scripts
--------------------------------------
-This image supports the [Source-to-Image](https://docs.openshift.com/container-platform/3.11/creating_images/s2i.html)
-(S2I) strategy in OpenShift. The Source-to-Image is an OpenShift framework
-which makes it easy to write images that take application source code as
-an input, use a builder image like this Python container image, and produce
-a new image that runs the assembled application as an output.
-
-To support the Source-to-Image framework, important scripts are included in the builder image:
-
-* The `/usr/libexec/s2i/assemble` script inside the image is run to produce a new image with the application artifacts.
-The script takes sources of a given application and places them into appropriate directories inside the image.
-It utilizes some common patterns in Perl application development (see the **Environment variables** section below).
-* The `/usr/libexec/s2i/run` script is set as the default command in the resulting container image (the new image with the application artifacts).
-It runs your application according to settings in `APP_MODULE`, `APP_FILE` or `APP_SCRIPT` environment variables or it tries to detect the best
-way automatically.
-
-Building an application using a Dockerfile
-------------------------------------------
-Compared to the Source-to-Image strategy, using a Dockerfile is a more
-flexible way to build a Python container image with an application.
-Use a Dockerfile when Source-to-Image is not sufficiently flexible for you or
-when you build the image outside of the OpenShift environment.
-
-To use the Python image in a Dockerfile, follow these steps:
-#### 1. Pull a base builder image to build on
+The full container image is an universal base image to build your containerized applications on top of it. However, its universal nature
+means that the resulting containers it produces consume a lot of disk space. This is caused mainly by the fact that the image contains
+npm, compilers, header files and some other packages one might need to install and deploy their applications.
 
-```
-podman pull registry.access.redhat.com/ubi8/python-39
-```
+Because size does matter for us and our customers, we have prepared this minimal container image with very limited subset
+of installed packages. There are no compilers, no header files, no npm etc and dnf tool is replaced with minimalistic
+reimplementation called microdnf making the resulting container images much smaller. It also causes some limitations
+but we also provide ways how to workaround them.
```

Comment:
Really yum? I mean, we are creating something new and we are building it on top of RHEL8 where we already have dnf so mentioning yum here seems to me like skipping one part of the yum - dnf - microdnf chain. I also think that the minimal container and containers, in general, are for modern people not for the ones who still use RHEL 5/6/7.

Reply:
There's been a marketing decision that the package manager in RHEL is supposed to be called `yum` even in RHEL 8, so that customers don't feel like they have to "switch" in some way. It's just a symlink to `dnf` and works the same, but `yum` is the official name.

See e.g. our official documentation: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/assembly_installing-and-using-python_configuring-basic-system-settings

Reply:
Oukej, oukej, old man :smile:

Reply:
Things should stay just as I like them! :rofl: